### PR TITLE
Policy

### PR DESCRIPTION
**Pull request by spinal84**
Rename and modify policy files according to API docs.

### CODE REVIEW

File: policy/policy_always_online.c [hunk 1]
```diff
@@ -296,136 +316,205 @@ network_type_notify_cb(GConfClient *client, guint cnxn_id, GConfEntry *entry,
...
     }
   }
 
-  if (!data->check_connections_timeout_id)
+  if (!data->count_iaps_id)
   {
-    data->check_connections_timeout_id =
-        g_timeout_add(500, check_connections_cb, data);
+    data->count_iaps_id =
+        g_timeout_add(POLICY_ALWAYS_ONLINE_IAP_TIMEOUT, policy_always_online_check, data);
```
Comment (freemangordon) [inline policy/policy_always_online.c]: more than 80 symbols per line, breaks coding style

File: policy/policy_always_online.c [hunk 2]
```diff
@@ -296,136 +316,205 @@ network_type_notify_cb(GConfClient *client, guint cnxn_id, GConfEntry *entry,
...
 
-  if (gconf_entry_get_value(entry) && !data->check_connections_timeout_id)
+  if (gconf_entry_get_value(entry) && !data->count_iaps_id)
   {
     ILOG_DEBUG("always online will soon check whether connections were added");
 
-    data->check_connections_timeout_id =
-        g_timeout_add(500, check_connections_cb, data);
+    data->count_iaps_id =
+        g_timeout_add(POLICY_ALWAYS_ONLINE_IAP_TIMEOUT, policy_always_online_check, data);
```
Comment (freemangordon) [inline policy/policy_always_online.c]: ditto

File: policy/policy_always_online.c [hunk 3]
```diff
@@ -296,136 +316,205 @@ network_type_notify_cb(GConfClient *client, guint cnxn_id, GConfEntry *entry,
...
-    data->get_device_mode_pending = 0;
-  }
+  policy_always_online_cancel_pending(data);
 
-  if (data->devmode_filter_set)
+  if (data->flightmode_signals)
     icd_dbus_disconnect_system_bcast_signal(MCE_SIGNAL_IF,
-                                            device_mode_ind_filter, data,
-                                            "member='" MCE_DEVICE_MODE_SIG "'");
+                                            policy_always_online_flightmode_sig, data,
```
Comment (freemangordon) [inline policy/policy_always_online.c]: ditto. multi-line if() is missing curly braces

File: policy/policy_any.c [hunk 4]
```diff
@@ -183,10 +185,9 @@ scan_cb(const guint status, const gchar *service_name,
             network->network_id = g_strdup(network_id);
             network->network_priority = network_priority;
             network->signal = signal;
-            data->network_list =
-                g_slist_insert_sorted(data->network_list, network,
-                                      (GCompareFunc)network_compare);
-            scan_data->data->iaps_found = TRUE;
+            data->found_networks =
+                g_slist_insert_sorted(data->found_networks, network, policy_any_sort_network);
```
Comment (freemangordon) [inline policy/policy_any.c]: more than 80 characters

File: policy/policy_any.c [hunk 5]
```diff
@@ -278,229 +279,248 @@ get_max_priority(const GSList *existing_requests)
...
-                                              data->network_types);
+    gchar *type = (gchar *)data->scan_types_list->data;
+    data->scan_types_list = g_slist_delete_link(data->scan_types_list,
+                                              data->scan_types_list);
 
     if (type && *type)
     {
-      struct scan_data *scan_data;
-
+      struct policy_scan_data *scan_data = policy_any_scan_data_new(
```
Comment (freemangordon) [inline policy/policy_any.c:525]: not really a good idea to mix renaming to code changes, it becomes very hard to review

File: policy/policy_iap_ask.c [hunk 6]
```diff
@@ -123,20 +125,20 @@ show_conn_dlg_cb(DBusPendingCall *pending, void *user_data)
...
   *(data->private) = g_slist_remove((GSList *)*(data->private), data);
   g_free(data);
 }
 
 static void
-icd_policy_ask_request_new(struct icd_policy_request *new_request,
-                           const GSList *existing_requests,
-                           icd_policy_request_new_cb_fn policy_done_cb,
-                           gpointer policy_token, gpointer *private)
+policy_iap_ask_request(struct icd_policy_request *new_request,
```
Comment (freemangordon) [inline policy/policy_iap_ask.c:135]: coding style?

File: policy/policy_change.c [hunk 7]
```diff
@@ -8,134 +8,161 @@
...
 static void
-icd_policy_change_destruct(gpointer *private)
+policy_change_destruct(gpointer *private)
 {
   struct policy_change_data *data = (struct policy_change_data *)*private;
 
-  icd_dbus_disconnect_system_bcast_signal(ICD_UI_DBUS_INTERFACE, change_cb,
-                                          data, ICD_UI_FILTER_CHANGE_SIG);
-  cancel_pending_call(data);
+  icd_dbus_disconnect_system_bcast_signal(ICD_UI_DBUS_INTERFACE, policy_change_done,
```
Comment (freemangordon) [inline policy/policy_change.c]: 80 chars

File: policy/policy_change.c [hunk 8]
```diff
@@ -8,134 +8,161 @@
...
   else
     ILOG_WARN("policy change dialog confirmed but actual dialog not implemented; bug in dialogs!");
 
-  policy_done(ICD_POLICY_REJECTED, data);
+  policy_change_do_cb(ICD_POLICY_REJECTED, data);
   dbus_message_unref(message);
 }
 
+static gboolean
+policy_change_confirm(struct policy_change_data *data)
```
Comment (freemangordon) [inline policy/policy_change.c:138]: merging renaming with code changes

File: policy/policy_change.c [hunk 9]
```diff
@@ -233,9 +238,9 @@ icd_policy_init(struct icd_policy_api *policy_api,
...
   struct policy_change_data *data = g_new0(struct policy_change_data, 1);
 
-  policy_api->new_request = icd_policy_change_new_request;
+  policy_api->new_request = policy_change_new_request;
   policy_api->private = data;
-  policy_api->destruct = icd_policy_change_destruct;
-  icd_dbus_connect_system_bcast_signal(ICD_UI_DBUS_INTERFACE, change_cb, data,
-                                       ICD_UI_FILTER_CHANGE_SIG);
+  policy_api->destruct = policy_change_destruct;
+  icd_dbus_connect_system_bcast_signal(ICD_UI_DBUS_INTERFACE, policy_change_done, data,
```
Comment (freemangordon) [inline policy/policy_change.c]: 80 chars per line

**Comment by freemangordon** [comment]
not finished with the review yet, don't make a new pull request :)

**Comment by spinal84** [comment]
Updated. No lines > 80 chars now.  
Renaming of files is just the first commit. So I added modifications according to API docs not to make more pull requests dedicated to the policy changes.

**Comment by spinal84** [comment]
About mixing internal names renaming with code changes. You are right.  
I tried to change the sources to make them like described in the API docs.
Should I redo the work to make separate commits for renaming of everything?

**Comment by freemangordon** [comment]
yes, please, do at least 2 commits per file - one for renaming of the functions, one for renaming of the structure and members and one more if there are code changes - like moving block of code in a new function. Sorry about that, but it is really hard to review what's going on if all is merged in a single commit per file.

**Comment by spinal84** [comment]
Please check pull request #8

**Pull request closed without merging**